bson/binary.py

```diff
@@ -549,6 +581,33 @@ def subtype(self) -> int:
         """Subtype of this binary data."""
         return self.__subtype
 
+    def as_numpy_vector(self) -> BinaryVector:
+        """From the Binary, create a BinaryVector where data is a 1-dim numpy array.
+        dtype still follows our typing (BinaryVectorDtype),
+        and padding is as we define it, notably equivalent to a negative value of count
+        in `numpy.unpackbits <https://numpy.org/doc/stable/reference/generated/numpy.unpackbits.html>`_.
+
+        :return: BinaryVector
+
+        .. versionadded:: 4.16
+        """
+        if self.subtype != VECTOR_SUBTYPE:
+            raise ValueError(f"Cannot decode subtype {self.subtype} as a vector")
+        if not _NUMPY_AVAILABLE:
+            raise ImportError("Converting binary to numpy.ndarray requires numpy to be installed.")
+        dtype, padding = struct.unpack_from("<sB", self, 0)
+        dtype = BinaryVectorDtype(dtype)
+
+        if dtype == BinaryVectorDtype.INT8:
+            data = np.frombuffer(self[2:], dtype="int8")
+        elif dtype == BinaryVectorDtype.FLOAT32:
+            data = np.frombuffer(self[2:], dtype="float32")
+        elif dtype == BinaryVectorDtype.PACKED_BIT:
+            data = np.frombuffer(self[2:], dtype="uint8")
```

Comment:
I don't think we're follow the rules of the spec for validating PACKED_BIT here (eg the validation in as_vector).

Reply:
I just added the same validations applied in `as_vector`.